ui/components/component-library/modal-focus/modal-focus.stories.tsx

```diff
@@ -0,0 +1,158 @@
+import React from 'react';
+import { ComponentStory, ComponentMeta } from '@storybook/react';
+
+import Box from '../../ui/box';
+
+import {
+  BorderColor,
+  DISPLAY,
+  FLEX_DIRECTION,
+} from '../../../helpers/constants/design-system';
+
+import { ModalFocus } from './modal-focus';
+import README from './README.mdx';
+
+export default {
+  title: 'Components/ComponentLibrary/ModalFocus',
+  component: ModalFocus,
+  parameters: {
+    docs: {
+      page: README,
+    },
+  },
+  args: {
+    children: (
+      <>
+        <p>Modal focus children</p>
+        <input />
+        <p>
+          Use the keyboard to try tabbing around you will notice that the focus
+          is locked to the content within modal focus
+        </p>
+      </>
+    ),
+  },
+} as ComponentMeta<typeof ModalFocus>;
+
+const Template: ComponentStory<typeof ModalFocus> = (args) => {
+  const [open, setOpen] = React.useState(false);
+  return (
+    <>
+      <button onClick={() => setOpen(true)}>Open</button>
```

Comment:
I used basic JSX here to reduce the amount of lines of code for the code examples and our Button component doesn't forward ref so some of these stories wouldn't have worked will make sure to create some issue to forwardReds on buttons